app/lib/admin/backend.dart

```diff
@@ -247,4 +253,58 @@ class AdminBackend {
       await tx.commit();
     });
   }
+
+  /// Handles GET '/api/admin/packages/<package>/assigned-tags'
+  ///
+  /// Note, this API end-point is intentioanlly locked down even if it doesn't
+  /// return anything secret. This is because the /admin/ section is only
+  /// intended to be exposed to administrators. Users can read the assigned-tags
+  /// through API that returns list of package tags.
+  Future<api.AssignedTags> handleGetAssignedTags(
+    String packageName,
+  ) async {
+    await _requireAdminPermission(AdminPermission.manageAssignedTags);
+    final package = await packageBackend.lookupPackage(packageName);
+    if (package == null) {
+      throw NotFoundException.resource(packageName);
+    }
+
+    return api.AssignedTags(
+      assignedTags: package.assignedTags,
+    );
+  }
+
+  /// Handles PUT '/api/admin/packages/<package>/assigned-tags'
```

Comment:
I'd find it better to have this as a `POST` API, with the request body
````
{
  "addedTags": [],
  "removedTags": [],
}
````

It would reduce the chance of accidentally removing (or adding) tags that the caller is removing (or adding) via bad pattern matching.

Reply:
I was very close to doing this... I'm onboard with this change if you also think it's sane :)